omegaconf/basecontainer.py

```diff
@@ -88,39 +88,41 @@ def __repr__(self) -> str:
 
     # Support pickle
     def __getstate__(self) -> Dict[str, Any]:
-        from omegaconf import DictConfig
+        from omegaconf import DictConfig, _utils
 
         dict_ = copy.copy(self.__dict__)
-        if sys.version_info < (3, 7, 0):  # pragma: no cover
-            ref_type = dict_["_metadata"].ref_type
-            if isinstance(self, DictConfig):
-                if ref_type is not Any:
-                    dict_["_metadata"].ref_type = Dict
-            else:
-                if ref_type is not Any:
-                    dict_["_metadata"].ref_type = List
+        ref_type = dict_["_metadata"].ref_type
+        if isinstance(self, DictConfig):
+            if ref_type is not Any:
+                key_type, element_type = _utils.get_dict_key_value_types(ref_type)
+                print(ref_type, key_type, element_type)
+                if key_type == Union[str, Enum]:
+                    key_type = None
```

Comment:
Is this decoded properly on the way out? looks like it will become `Dict[Any, ...]`.
The reason you missed it is because you added logic that is not covered by your tests.

Reply:
Internally is Dict[Any, Any] but it behaves like Dict[Union[str, Enum], Any].

Reply:
If you do get_ref_type it returns the `Dict[Union[str, Enum], Any]`